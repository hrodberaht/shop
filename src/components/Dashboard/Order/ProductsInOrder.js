import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ProductsInOrder extends Component {
  render() {
    const {
      order: { productsOrder },
    } = this.props;
    return (
      <td colSpan="5">
        <table>
          <tbody>
            <tr>
              <th>Name:</th>
              <th>Pcs:</th>
              <th>Price</th>
              <th>Total price:</th>
            </tr>
            {productsOrder.map((product) => {
              const {
                id, name, price, pcsOrder, totalPrice,
              } = product;
              return (
                <tr key={id}>
                  <td>{name}</td>
                  <td>{pcsOrder}</td>
                  <td>{price}</td>
                  <td>{totalPrice}</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </td>
    );
  }
}

ProductsInOrder.propTypes = {
  order: PropTypes.objectOf(PropTypes.arrayOf).isRequired,
};
