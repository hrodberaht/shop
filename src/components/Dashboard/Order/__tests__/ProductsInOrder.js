import ProductsInOrder from '../ProductsInOrder';

const defaultProps = {
  order: {
    productsOrder: [
      {
        productId: 'product-1',
        name: 'ZX3',
        price: 400,
        pcsOrder: 1,
        totalPrice: 400,
        id: 'Ld6GsMELf',
      },
    ],
  },
};
const setup = buildSetup(ProductsInOrder, defaultProps);

describe('<ProductsInOrder />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
