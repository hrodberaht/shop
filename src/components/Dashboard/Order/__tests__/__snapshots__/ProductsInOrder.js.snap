// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsInOrder /> should render without crash 1`] = `
<td
  colSpan="5"
>
  <table>
    <tbody>
      <tr>
        <th>
          Name:
        </th>
        <th>
          Pcs:
        </th>
        <th>
          Price
        </th>
        <th>
          Total price:
        </th>
      </tr>
      <tr
        key="Ld6GsMELf"
      >
        <td>
          ZX3
        </td>
        <td>
          1
        </td>
        <td>
          400
        </td>
        <td>
          400
        </td>
      </tr>
    </tbody>
  </table>
</td>
`;
